### PR TITLE
opcodes, ontalk handling in instances, build fix (maybe)

### PR DESCRIPTION
**Pull request by GokuWeedLord**

### CODE REVIEW

File: src/servers/sapphire_zone/Actor/Actor.cpp [hunk 1]
```diff
@@ -313,6 +313,12 @@ void Core::Entity::Actor::setCurrentZone( ZonePtr currZone )
    m_pCurrentZone = currZone;
 }
 
+/*! \return InstanceContentPtr to the current instance, null if not an instance or not set */
+Core::InstanceContentPtr Core::Entity::Actor::getCurrentInstance() const
+{
+   return getCurrentZone()->getAsInstanceContent();
```
Comment (takhlaq) [inline src/servers/sapphire_zone/Actor/Actor.cpp]: hmm wouldn't we want to check if they're loaded into a zone trying to access it

Comment (SapphireMordred) [reply src/servers/sapphire_zone/Actor/Actor.cpp]: I don't disagree. Casting into instanceContent without knowing it actually is one may be problematic.

Comment (GokuWeedLord) [reply src/servers/sapphire_zone/Actor/Actor.cpp]: dynamic_pointer_cast checks the type first, if it doesn't match it'll just return nullptr

Comment (GokuWeedLord) [reply src/servers/sapphire_zone/Actor/Actor.cpp]: rather, if the types are incompatible it'll be nullptr

Comment (takhlaq) [reply src/servers/sapphire_zone/Actor/Actor.cpp]: looks like my phone ate some words, i meant check their current zone isnt actually nullptr before trying to call getAsInstance()

Comment (GokuWeedLord) [reply src/servers/sapphire_zone/Actor/Actor.cpp]: ah yeah, that could be a problem if it's used elsewhere, will fix

**Pull request merged**